### PR TITLE
Added support for file properties in Tiled maps

### PR DESCRIPTION
**Pull request by tylerhasman**
I need to have file properties in my tiled map to link map objects to script files.
I'm sure that this isn't the best way to do it but I am open for feedback.

### CODE REVIEW

**Comment by EtK2000** [comment]
I'm not really sure the value of this, but if it's already implemented you might want to override toString() to return the file path.

**Comment by tylerhasman** [comment]
It's useful because my game scripts are in Javascript and the tiled editor has no built-in formatter. The scripts get quite messy because of this so I have chosen to migrate the longer ones to files.

**Comment by tylerhasman** [comment]
> I'm not really sure the value of this, but if it's already implemented you might want to override toString() to return the file path.

To add one to my above reply:
When you try to load a tiled map with a file attribute, it throws an exception so this is a necessary change.

**Comment by EtK2000** [comment]
I don't really use the classes that you changed so I don't understand the addition and if/why it's necessary.
I'll step away from this and let someone more knowledgeable look at it.

**Comment by tylerhasman** [comment]
The new Tiled update supports file properties so it only makes sense for LibGDX to handle them.

**Comment by MobiDevelop** [comment]
I'm sure the maps stuff is so far behind the Tiled releases that there's plenty more that needs to be done besides handling file properties.

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[1.9.10]
+- Added support for 'file' properties when loading Tiled maps.
```
Comment (MobiDevelop) [inline CHANGES]: Maybe add a space to separate from the 1.9.9 list.

File: gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java [hunk 2]
```diff
@@ -351,6 +354,8 @@ private Object castProperty (String name, String value, String type) {
 			String opaqueColor = value.substring(3);
 			String alpha = value.substring(1, 3);
 			return Color.valueOf(opaqueColor + alpha);
+		}else if(type.equals("file")){
```
Comment (MobiDevelop) [inline gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java]: Spacing is off on this.

File: gdx/src/com/badlogic/gdx/maps/tiled/TmxMapLoader.java [hunk 3]
```diff
@@ -77,6 +77,7 @@ public TiledMap load (String fileName, TmxMapLoader.Parameters parameters) {
 			this.convertObjectToTileSpace = parameters.convertObjectToTileSpace;
 			this.flipY = parameters.flipY;
 			FileHandle tmxFile = resolve(fileName);
+			this.tmxFile = tmxFile;
```
Comment (MobiDevelop) [inline gdx/src/com/badlogic/gdx/maps/tiled/TmxMapLoader.java]: Should probably just set the field in the previous line.

File: gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java [hunk 4]
```diff
@@ -133,6 +133,7 @@ public TiledMap load (String fileName, AtlasTiledMapLoaderParameters parameter)
 			}
 
 			FileHandle tmxFile = resolve(fileName);
+			this.tmxFile = tmxFile;
```
Comment (MobiDevelop) [inline gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: Should probably just set the field in the previous line.

**Comment by MobiDevelop** [comment]
After refactor, this is no longer able to be merged due to conflicts.

**Comment by MrStahlfelge** [comment]
@tylerhasman are you interested in updating the PR? otherwise we should close it due to the mentioned problem.

**Comment by tylerhasman** [comment]
@MrStahlfelge i'll see if I have time to work on it

**Pull request closed without merging**